src/java/com/puppetlabs/ssl_utils/SSLUtils.java

```diff
@@ -737,8 +737,12 @@ else if (obj instanceof PEMKeyPair)
     {
         List<Object> objects = pemToObjects(reader);
         List<PrivateKey> results = new ArrayList<PrivateKey>(objects.size());
-        for (Object o : objects)
-            results.add(objectToPrivateKey(o));
+        for (Object o : objects) {
+            // Filter out EC params
+            if (!(o instanceof ASN1ObjectIdentifier)) {
```

Comment:
Nice approach, way better than the idea I had about filtering the string before passing it to the parser.

Is it possible to make the conditional a positive on the instance of `PrivateKey`, rather than a negative condition on the Identifier?

Reply:
I was going to do that, but I noticed that `objectToPrivateKey` has an else block that throws a nice error if it gets something that isn't a private key. Having this be a positive check would lose that.

I think I can preserve both, by moving the error handling into this function and special-casing `ASN1ObjectIdentifier`, though.

Reply:
`objectToPrivateKey` is available for calling outside of `pemToPrivateKeys`, so I left the checking there, even though it's sometimes redundant now. I didn't want to ignore this object in that function, since the EC param issue is only relevant when parsing PEMs.

Still not totally sure if the new version is better...

Reply:
If we don't care about throwing when the thing in the PEM is not a private key, I can make this much simpler. But I feel like that feedback might be useful?

Reply:
Mmmm... I don't know that I care about throwing that much. Honestly I think logging and not throwing is probably better, but I don't know if we have other things depending on that functionality.

I do think, unfortunately, I would prefer the slightly abnormal conditional to duplicating throws.

Reply:
Okay. I'll put this back for now. We don't currently have any logging at the Java level. That could probably be remedied, but is maybe outside the scope of this work.